detox/src/client/actions/SyncStatusFormatter.js

```diff
@@ -0,0 +1,277 @@
+const Ajv = require('ajv');
+
+const DetoxRuntimeError = require('../../errors/DetoxRuntimeError');
+
+class SyncStatusFormatter {
+  static formatJSONStatus(jsonStatus) {
+    if (!SyncStatusFormatter.isValidJSONStatus(jsonStatus)) {
+      throw new DetoxRuntimeError(`Given sync status is not compatible with the status schema, ` +
+        `given status: ${JSON.stringify(jsonStatus)}`);
+    }
+
+    if (SyncStatusFormatter.isAppIdle(jsonStatus)) {
+      return 'The app seems to be idle';
+    }
+
+    if (jsonStatus.busy_resources === undefined) {
+      throw new DetoxRuntimeError(`Given sync status is invalid, app is busy but busy-resources are not defined`);
+    }
+    const resourcesDescriptions = SyncStatusFormatter.resourcesDescriptionsFromJSON(jsonStatus.busy_resources);
+    return `The app is busy with the following tasks:\n${resourcesDescriptions.join('\n')}`;
+  }
+
+  static isValidJSONStatus(jsonStatus) {
+    const ajv = new Ajv();
+    const isValidStatus = ajv.compile(SyncStatusFormatter.jsonStatusSchema());
+    return isValidStatus(jsonStatus);
```

Comment:
Why don't you compile it in advance to use just the function, `isValidStatus`?

Reply:
You're right, this should be done outside in advance. Nice catch!